src/System.Security.Permissions/src/System/ApplicationIdentity.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Runtime.Serialization;
+using System.Security;
+
+namespace System
+{
+    [Serializable]
+    public sealed class ApplicationIdentity : ISerializable
+    {
+        private ApplicationIdentity() { }
+        [SecurityCritical]
+        private ApplicationIdentity(SerializationInfo info, StreamingContext context) { }
+
+        [SecuritySafeCritical]
+        public ApplicationIdentity(string applicationIdentityFullName) { }
+
+        public string FullName { get { return null; } }
+
+        public string CodeBase { get { return null; } }
+
+        public override string ToString() { return null; }
+
+        /// <internalonly/>
+        [SecurityCritical]  // auto-generated_required
```

Comment:
Therei s discussion here of where these are needed https://github.com/dotnet/corefx/issues/12592#issuecomment-253971384 
This is not an out of box (OOB) shipping assembly so we should not need security annotations except [SecurityCritical] and that only in ref , not src.